db/migrate/20230730013646_allow_non_unique_deleted_user_emails.rb

```diff
@@ -0,0 +1,12 @@
+class AllowNonUniqueDeletedUserEmails < ActiveRecord::Migration[6.1]
+  # allow non-unique emails for deleted users
+  def change
+    # assuming users table isn't huge, building this index should be quick so
+    # it's probably okay to lock the table
+    StrongMigrations.disable_check(:add_index)
```

Comment:
Oh, this is a wonderful thing.  For whatever reason I never saw this in the documentation, and adding indexes seems to hang terribly on some installations.  Thanks!